### PR TITLE
New `tag_ami_with_result` option

### PR DESCRIPTION
**Pull request by fruch**
with this new option, we tag the test final results ontop of the scylla ami being used
this way we could keep tab of what is the status regarding this AMI,
and if it's safe enough to publish

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -1672,3 +1672,14 @@ def list2reason(exc_list):
             TEST_LOG.error("-" * 70)
             TEST_LOG.error(text)
             TEST_LOG.error("-" * 70)
+
+        self.tag_ami_with_result(error, failure)
+
+    def tag_ami_with_result(self, test_error, test_failure):
+        if self.params('cluster_backend', '') == 'aws' and self.params('tag_ami_with_result'):
+            test_result = 'PASSED' if not test_error and not test_failure else 'ERROR' if test_error else 'FAILURE'
```
Comment (bentsi) [inline sdcm/tester.py]: this is too long and unreadable, convert to regular if

Comment (fruch) [reply sdcm/tester.py]: fixed

File: sdcm/tester.py [hunk 2]
```diff
@@ -1682,6 +1682,8 @@ def list2reason(exc_list):
             TEST_LOG.error(text)
             TEST_LOG.error("-" * 70)
 
+        self.tag_ami_with_result(error, failure)
```
Comment (bentsi) [inline sdcm/tester.py]: this is not the place to put this. Move to teardown and make sure if it fails it doesn't fail the whole teardown sequence

Comment (fruch) [reply sdcm/tester.py]: done

**Comment by bentsi** [comment]
please fix conflicts

**Comment by bentsi** [comment]
please resolve conflicts

**Comment by bentsi** [comment]
???

**Comment by fruch** [comment]
@bentsi conflicts resolved

**Pull request merged**